sale_measure/measure.py

```diff
@@ -0,0 +1,82 @@
+from openerp import fields, models, api
+from collections import defaultdict
+from openerp import exceptions
+from openerp.osv import orm
+from lxml import etree
+
+
+class MeasureMeasure(models.Model):
+    _name = 'measure.measure'
+    _description = "Measure for each partner"
+
+    product_id = fields.Many2one(
+        'product.product',
+        string="Product",
+        required=True)
+    partner_id = fields.Many2one(
+        'res.partner',
+        string="partner",
+        required=True)
+    measure_form_type = fields.Selection(
+        related='product_id.measure_form_type',
+        readonly=True)
+    sale_line_ids = fields.One2many(
+        'sale.order.line',
+        'measure_id',
+        string="Measure Ligne")
+    qty = fields.Float('Quantity')
+
+    @classmethod
+    def get_form(cls):
+        return {}
+
+    def _check_form(self):
+        for key, value in zip(self.get_form()[self.measure_form_type].keys(),
+                              self.get_form()[self.measure_form_type].values()):
+            if 'value' in value.keys():
+                if self[key] not in value['value']:
+                    raise exceptions.Warning(
```

Comment:
It's not a clean way to inject the variable in your message, search for exemple in odoo